src/network/networkd-address.c

```diff
@@ -1001,3 +1006,163 @@ int prefix_new_static(Network *network, unsigned section, Prefix **ret) {
+                return 0;
+        }
+
+        if (sd_radv_prefix_set_prefix(p->radv_prefix, &buffer.in6, prefixlen) < 0)
+                return -EADDRNOTAVAIL;
+
+        log_syntax(unit, LOG_INFO, filename, line, r, "Found prefix %s with length %d", address, prefixlen);
+
+        p = NULL;
+
+        return 0;
+}
+
+int config_parse_prefix_flags(const char *unit,
+                              const char *filename,
+                              unsigned line,
+                              const char *section,
+                              unsigned section_line,
+                              const char *lvalue,
+                              int ltype,
+                              const char *rvalue,
+                              void *data,
+                              void *userdata) {
+        Network *network = userdata;
+        _cleanup_prefix_free_ Prefix *p = NULL;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = prefix_new_static(network, section_line, &p);
+        if (r < 0)
+                return r;
+
+        r = parse_boolean(rvalue);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse address flag, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        if (streq(lvalue, "OnLink")) {
+                if (sd_radv_prefix_set_onlink(p->radv_prefix, r) < 0)
+                        return -EINVAL;
+        } else if (streq(lvalue, "AddressAutoconfiguration")) {
+                if (sd_radv_prefix_set_address_autoconfiguration(p->radv_prefix, r) < 0)
+                        return -EINVAL;
+        }
+
+        p = NULL;
+
+        return 0;
+}
+
+int config_parse_prefix_uint32(const char *unit,
+                               const char *filename,
+                               unsigned line,
+                               const char *section,
+                               unsigned section_line,
+                               const char *lvalue,
+                               int ltype,
+                               const char *rvalue,
+                               void *data,
+                               void *userdata) {
+        Network *network = userdata;
+        _cleanup_prefix_free_ Prefix *p = NULL;
+        uint32_t i;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = prefix_new_static(network, section_line, &p);
+        if (r < 0)
+                return r;
+
+        r = safe_atou(rvalue, &i);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Lifetime is invalid, ignoring assignment: %s", rvalue);
+                return 0;
+        }
```

Comment:
shouldn't be some checking of the valid range of the lifetime value?

in what uniot is the lifetime? if this is in time units, then this should really be using parse_sec() and the setting should be called PreferredLifetimeSec to follow our usual naming schemes